crates/ruff/src/rules/refurb/rules/repeated_append.rs

```diff
@@ -0,0 +1,377 @@
+            // we should mark it as "non-consecutive".
+            if group.appends.len() > 1 {
+                group.is_consecutive = false;
+            }
+            last_binding = binding;
+        }
+    }
+
+    map.into_values()
+        // Groups with just one element are not interesting for us.
+        .filter(|group| group.appends.len() > 1)
+        .collect()
+}
+
+#[inline]
+fn get_binding_stmt(group: &AppendGroup) -> NodeId {
+    group.appends.first().unwrap().binding.source.unwrap()
+}
+
+#[inline]
+fn get_or_add<'a, 'b>(
+    map: &'b mut HashMap<NodeId, AppendGroup<'a>>,
+    append: &Append<'a>,
+) -> &'b mut AppendGroup<'a> {
+    let binding_stmt = append.binding.source.unwrap();
+    let group = map.entry(binding_stmt).or_insert(AppendGroup {
+        appends: vec![],
+        is_consecutive: true,
+    });
+    group.appends.push(append.clone());
+    group
+}
+
+fn make_suggestion(group: &AppendGroup, generator: Generator) -> String {
+    let appends = &group.appends;
+
+    assert!(!appends.is_empty());
+    let first = appends.first().unwrap();
+
+    assert!(appends
+        .iter()
+        .all(|append| append.binding.source == first.binding.source));
+
+    // Here we construct `var.extend((elt1, elt2, ..., eltN))
+    //
+    // Each eltK comes from an individual `var.append(eltK)`.
+    let elts: Vec<ast::Expr> = appends
+        .iter()
+        .map(|append| append.argument.clone())
+        .collect();
+    // Join all elements into a tuple: `(elt1, elt2, ..., eltN)`
+    let tuple = ast::ExprTuple {
+        elts,
+        ctx: ast::ExprContext::Load,
+        range: TextRange::default(),
+    };
+    // Make `var.extend`.
+    // NOTE: receiver is the same for all appends and that's why we can take the first.
+    let attr = ast::ExprAttribute {
+        value: Box::new(first.receiver.clone().into()),
+        attr: ast::Identifier::new("extend".to_string(), TextRange::default()),
+        ctx: ast::ExprContext::Load,
+        range: TextRange::default(),
+    };
+    // Make the actual call `var.extend((elt1, elt2, ..., eltN))`
+    let call = ast::ExprCall {
+        func: Box::new(attr.into()),
+        arguments: ast::Arguments {
+            args: vec![tuple.into()],
+            keywords: vec![],
+            range: TextRange::default(),
+        },
+        range: TextRange::default(),
+    };
+    // And finally, turn it into a statement.
+    let stmt = ast::StmtExpr {
+        value: Box::new(call.into()),
+        range: TextRange::default(),
+    };
+    generator.stmt(&stmt.into())
+}
+
+fn match_append<'a>(checker: &'a Checker, stmt: &'a Stmt) -> Option<Append<'a>> {
+    let Stmt::Expr(ast::StmtExpr { value, .. }) = stmt else {
+        return None;
+    };
+
+    let Expr::Call(ast::ExprCall {
+        func, arguments, ..
+    }) = value.as_ref()
+    else {
+        return None;
+    };
+
+    if arguments.args.len() != 1 {
```

Comment:
We'd generally prefer:

```rust
let [arg] = arguments.args.as_slice() else {
    return;
}
```

Which lets us avoid the unwrap further down.

Reply:
Yep, I started using this pattern only later 😅 thanks!